src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -114,33 +112,28 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
                     continue;
                 }
 
-                if (parameterDescription.Source == BindingSource.Body)
-                {
-                    hasJsonBody = true;
-                }
-
                 apiDescription.ParameterDescriptions.Add(parameterDescription);
             }
 
-            // Get custom attributes for the handler. ConsumesAttribute is one of the examples. 
-            var acceptsRequestType = routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>()?.RequestType;
-            if (acceptsRequestType is not null)
+            // Get IAcceptsMetadata.
+            var acceptsMetadata = routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>();
+            if (acceptsMetadata is not null)
             {
+                var acceptsRequestType = acceptsMetadata.RequestType;
+                var isOptional = acceptsMetadata.IsOptional;
                 var parameterDescription = new ApiParameterDescription
                 {
-                    Name = acceptsRequestType.Name,
-                    ModelMetadata = CreateModelMetadata(acceptsRequestType),
+                    Name = acceptsRequestType is not null ? acceptsRequestType.Name : typeof(void).Name,
+                    ModelMetadata = CreateModelMetadata(acceptsRequestType ?? typeof(void)),
```

Comment:
This is an interesting side effect of this change. The type isn't required right?

Reply:
Oh yes, an interesting one when you use ConsumesAttribute without a type. Accepts requires a type, so we are good on that one.  while testing with Swashbuckle, I noticed that it  won't take null values for some of the description properties (Type, Model, etc.). So passing a void type is away to mitigate that. Also, the user  can look at the open api result and quickly realize that what they declared is not correct. so, I am now convinced this is more helpful to the user.